src/rest/README.md

```diff
@@ -0,0 +1,496 @@
+      "relationships": {
+        "result": {
+          "data": {
+            "type": "diagnostics",
+            "id": "0a97ef16-1997-43dd-91c4-7fbcb1ec6713"
+          }
+        }
+      }
+    }
+  ],
+  "meta": {
+    "collection": {
+      "offset": 0,
+      "limit": 100,
+      "total": 1,
+      "pending": 0
+    }
+  }
+}
+```
+
+### Examples of general resource features
+
+The request must comprise a header field `Accept` set to `application/vnd.api+json` for JSON:API content format, or `application/json` for JSON content format in the response. Accordingly, the response comprises the header field `Content-type`.
+
+#### Get collection
+
+All items of the collection are gettable in a single request, for example:
+
+```
+curl -G http://<otbr-address>/api/<collectionName> -H 'Accept: application/vnd.api+json'
+```
+
+#### Get collection item by id
+
+Individual items are gettable from the collections, for example:
+
+```
+curl -G http://<otbr-address>/api/<collectionName>/<itemId> -H 'Accept: application/vnd.api+json'
+```
+
+#### Get items by type
+
+The collection can be filtered by type of the items, for example:
+
+```
+curl -G http://<otbr-address>/api/<collectionName>?fields[<type>] -H 'Accept: application/vnd.api+json'
+```
+
+#### Get items by type with limited attributes
+
+The collection can be filtered by type of the items, for example:
+
+```
+curl -G http://<otbr-address>/api/<collectionName>?fields[<type>]=attribute1,attribute2 -H 'Accept: application/vnd.api+json'
+```
+
+Detailed request examples are provided in the test folder as a "Bruno Request Collection". Use the free open-source API client [Bruno](https://www.usebruno.com/).
+
+## Key file description
+
+- `actions_list.cpp`, `actions_list.hpp`
+
+  Implements the action collection, derived from `BasicCollection`.
+
+- `commissioner_manager.cpp`, `commissioner_manager.hpp`
+
+  Handles starting and stopping the on-mesh commissioner as needed by the specific queued action items.
+
+- `network_diag_handler.cpp`, `network_diag_handler.hpp`
+
+  Handles collecting network diagnostic TLVs on top of OT core `netdiag` and `meshdiag` API.
+
+- `rest/actions/*`
+
+  Implement the action item stored in action collection. The action item is derived from `BasicActions` class, which is derived from `BasicCollectionItem`.
+
+- `rest_generic_collection.cpp`, `rest_generic_collection.hpp`
+
+  Implements the `BasicCollectionItem` and `BasicCollection` classes.
+
+- `rest_*_coll.cpp`, `rest_*_coll.hpp`
+
+  Implements the specific `devices` and `diagnostics` collection classes derived from `BasicCollection` class.
+
+- `uuid.cpp`,`uuid.hpp`
+
+  Implements functionality to handle UUID (generation, parse, unparse) used for collection items.
+
+- `tests/restjsonapi/*`
+
+  Test setup including implementation for various test cases.
+
+## How to rebuild and experience this work
+
+Follow these steps to install/build OTBR.
+
+1. Checkout this PR
```

Comment:
Does this step need to change after this is submitted?  Should it just be clone the repo?

Reply:
Good point, I added a link to the border router guide, and thus eliminated some steps here.